proofs.go

```diff
@@ -1,93 +1,168 @@
 package smt
 
-import(
-    "hash"
-    "bytes"
-    "errors"
+import (
+	"bytes"
+	"errors"
+	"hash"
+	"math"
 )
 
+// SparseMerkleProof is a Merkle proof for a element in a SparseMerkleTree.
+type SparseMerkleProof struct {
+	// SideNodes is an array of the sibling nodes leading up to the leaf of the proof.
```

Comment:
Isn't it down to the leaf of the proof?

Reply:
https://www.youtube.com/watch?v=Xc4xYacTu-E